packages/webapp/src/components/Table/v2/index.jsx

```diff
@@ -0,0 +1,282 @@
+import { BsThreeDots } from 'react-icons/bs';
+import clsx from 'clsx';
+import EnhancedTableHead from './TableHead';
+import Button from '../../Form/Button';
+import styles from './styles.module.scss';
+
+const More = ({ onClickLoadMore, onClickMore, invisibleRowCount, dense, colSpan }) => {
+  const { t } = useTranslation();
+
+  if (invisibleRowCount <= 0) {
+    return null;
+  }
+
+  if (onClickMore) {
+    return (
+      <TableRow>
+        <TableCell colSpan={colSpan} className={clsx(styles.tableCell, dense && styles.dense)}>
+          <button onClick={onClickMore} className={styles.moreButton}>
+            <BsThreeDots />
+            <span>{t('TABLE.NUMBER_MORE', { number: invisibleRowCount })}</span>
+          </button>
+        </TableCell>
+      </TableRow>
+    );
+  }
+
+  return (
+    <TableRow>
+      <TableCell colSpan={colSpan} className={clsx(styles.loadMoreCell)}>
+        <Button
+          className={styles.loadMoreButton}
+          onClick={onClickLoadMore}
+          color={'secondary'}
+          sm={true}
+        >
+          {t('TABLE.LOAD_MORE')}
+        </Button>
+      </TableCell>
+    </TableRow>
+  );
+};
+
+More.propTypes = {
+  onClickLoadMore: PropTypes.func,
+  onClickMore: PropTypes.func,
+  invisibleRowCount: PropTypes.number,
+  dense: PropTypes.bool,
+  colSpan: PropTypes.number,
+};
+
+function descendingComparator(a, b, orderBy) {
+  if (b[orderBy] < a[orderBy]) {
+    return -1;
+  }
+  if (b[orderBy] > a[orderBy]) {
+    return 1;
+  }
+  return 0;
+}
+
+function getComparator(order, orderBy) {
+  return order === 'desc'
+    ? (a, b) => descendingComparator(a, b, orderBy)
+    : (a, b) => -descendingComparator(a, b, orderBy);
+}
+
+/**
+ * A table component built utilizing the Material Ui Table.
+ * https://mui.com/material-ui/react-table/
+ */
+export default function EnhancedTable(props) {
+  const {
+    columns,
+    data,
+    FooterCell,
+    minRows = 10,
+    onRowClick,
+    showPagination,
+    pageSizeOptions = [5, 10, 20, 50],
+    onClickMore,
+    itemsToAddPerLoadMoreClick = 5,
+    dense = true,
+    shouldFixTableLayout = false,
+    defaultOrderBy = '',
+  } = props;
+
+  const [order, setOrder] = useState('asc');
+  const [orderBy, setOrderBy] = useState(defaultOrderBy);
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(minRows);
+
+  const fullColSpan = columns.reduce((total, column) => total + (column.id ? 1 : 0), 0);
+
+  const handleRequestSort = (event, property) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const handleRowClick = (event, index) => {
+    if (onRowClick) {
+      onRowClick(event, index);
+    }
+  };
+
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
```

Comment:
UI/UX question - Should this be reset to 0? Or should we try to keep the last record visible still visible?